frontend/src/modules/Catalog/components/GlossarySearchResultItem.js

```diff
@@ -320,7 +370,14 @@ export const GlossarySearchResultItem = ({ hit }) => {
               onClick={() =>
                 hit.resourceKind === 'dashboard'
                   ? handleRequestDashboardAccessModalOpen()
-                  : handleRequestAccessModalOpen()
```

Comment:
same approach

```
modal_open_func = {
'dashboard': handleRequestDashboardAccessModalOpen,
'dataset' : handleRequestAccessModalOpen,
...
}

```
`onClick = {modal_open_func[hit.resourceKind]}`

Reply:
This one was causing a lot of issues on rendering